lms/djangoapps/courseware/grades.py

```diff
@@ -25,10 +26,71 @@
 from .module_render import get_module_for_descriptor
 from submissions import api as sub_api  # installed from the edx-submissions repository
 from opaque_keys import InvalidKeyError
+from xblock.core import XBlock
 
 
 log = logging.getLogger("edx.courseware")
 
+GRADED_LOCATION_CATEGORIES = set(
+    cat for (cat, xblock_class) in XBlock.load_classes() if (
+        getattr(xblock_class, 'has_score', True) or getattr(xblock_class, 'has_children', True)
+    )
```

Comment:
so `xb_cls.has_score` does break, since there are some `xb_cls`s that don't have scores. I don't know if that's the case for `has_children` too--so I included this for safety. I'll check to see if `has_children` is always defined on xblock.